openvdb/points/PointMove.h

```diff
@@ -0,0 +1,997 @@
+        }
+    }
+
+private:
+    HandleArray mHandles;
+}; // struct AttributeHandles
+
+
+template <typename DeformerT, typename TreeT, typename FilterT>
+struct BuildMoveMapsOp
+{
+    using LeafT = typename TreeT::LeafNodeType;
+    using LeafArrayT = std::vector<LeafT*>;
+    using LeafManagerT = typename tree::LeafManager<TreeT>;
+
+    BuildMoveMapsOp(const DeformerT& deformer,
+                    GlobalPointIndexMap& globalMoveLeafMap,
+                    LocalPointIndexMap& localMoveLeafMap,
+                    const std::map<Coord, Index>& targetLeafMap,
+                    const math::Transform& targetTransform,
+                    const math::Transform& sourceTransform,
+                    const FilterT& filter)
+        : mDeformer(deformer)
+        , mGlobalMoveLeafMap(globalMoveLeafMap)
+        , mLocalMoveLeafMap(localMoveLeafMap)
+        , mTargetLeafMap(targetLeafMap)
+        , mTargetTransform(targetTransform)
+        , mSourceTransform(sourceTransform)
+        , mFilter(filter) { }
+
+    void operator()(LeafT& leaf, size_t idx) const
+    {
+        DeformerT deformer(mDeformer);
+        deformer.reset(leaf, idx);
+
+        // determine source leaf node origin and offset in the source leaf vector
+
+        Coord sourceLeafOrigin = leaf.origin();
+
+        auto sourceHandle = AttributeWriteHandle<Vec3f>::create(leaf.attributeArray("P"));
+
+        for (auto iter = leaf.beginIndexOn(mFilter); iter; iter++) {
+
+            const bool useIndexSpace = DeformerTraits<DeformerT>::IndexSpace;
+
+            // extract index-space position and apply index-space deformation (if applicable)
+
+            Vec3d positionIS = sourceHandle->get(*iter) + iter.getCoord().asVec3d();
+            if (useIndexSpace) {
+               deformer.apply(positionIS, iter);
+            }
+
+            // transform to world-space position and apply world-space deformation (if applicable)
+
+            Vec3d positionWS = mSourceTransform.indexToWorld(positionIS);
+            if (!useIndexSpace) {
+                deformer.apply(positionWS, iter);
+            }
+
+            // transform to index-space position of target grid
+
+            positionIS = mTargetTransform.worldToIndex(positionWS);
+
+            // determine target voxel and offset
+
+            Coord targetVoxel = Coord::round(positionIS);
+            Index targetOffset = LeafT::coordToOffset(targetVoxel);
+
+            // set new local position in source transform space (if point has been deformed)
+
+            Vec3d voxelPosition(positionIS - targetVoxel.asVec3d());
+            sourceHandle->set(*iter, voxelPosition);
+
+            // determine target leaf node origin and offset in the target leaf vector
+
+            Coord targetLeafOrigin = targetVoxel & ~(LeafT::DIM - 1);
+            assert(mTargetLeafMap.find(targetLeafOrigin) != mTargetLeafMap.end());
+            const Index targetLeafOffset(mTargetLeafMap.at(targetLeafOrigin));
```

Comment:
You're suggesting this so as to be able to use switch the std::map into a std::unordered_map? If so that's a modest but nonetheless worthwhile improvement. You do however still need to compute the origin for target <-> source leaf comparison.